### PR TITLE
Automatically generate the _toc file

### PR DESCRIPTION
**Pull request by guillim**
changing the Book structure in order to automatically generate the _toc file with the `jupyter-book toc docs --output-folder docs` command

### CODE REVIEW

**Comment by guillim** [comment]
Any updates here ?

**Comment by psorianom** [comment]
Yes @guillim ! let's adopt the automatic way! I added some commits to your PR to test different configs and reordered some files to make the book clearer. 
Just a little problem for the future, I could not find out how to sort the main "sections" of the notebook : "lab ia" , "piaf" , "csv-detective". I added a number at the beginning of the folder names (`1_lab_ia`, `2_piaf`, `3_csv-detective`) but no dice. We could look into that in the future :+1:

File: .gitignore [hunk 1]
```diff
@@ -131,3 +131,7 @@ dmypy.json
 # Docs
 _build
 
+/bin
```
Comment (psorianom) [inline .gitignore:134]: Why do we need to add these system folders in our gitignore ? Do you need to add them on your side?

Comment (guillim) [reply .gitignore:134]: I don't remember why. I think it has to do with an automatic generation of this file

**Comment by guillim** [comment]
Shall we merge @psorianom ?

**Comment by psorianom** [comment]
yes!

**Pull request merged**